weather_dl/download_pipeline/parsers.py

```diff
@@ -409,19 +402,17 @@ def require(condition: bool, message: str, error_type: t.Type[Exception] = Value
             """.format(num_template_replacements, num_partition_keys))
 
     # Ensure consistent lookup.
-    config['parameters']['partition_keys'] = partition_keys
+    config.partition_keys = partition_keys
 
     return config
 
 
 def prepare_target_name(config: Config) -> str:
     """Returns name of target location."""
-    parameters = config['parameters']
-    target_path = t.cast(str, parameters.get('target_path', ''))
-    partition_keys = t.cast(t.List[str],
-                            cp.copy(parameters.get('partition_keys', list())))
+    target_path = config.target_path
+    partition_keys = config.partition_keys
 
-    partition_dict = OrderedDict((key, typecast(key, config['selection'][key][0])) for key in partition_keys)
+    partition_dict = OrderedDict((key, typecast(key, config.selection[key][0])) for key in partition_keys)
     target = target_path.format(*partition_dict.values(), **partition_dict)
```

Comment:
Similar to the above point, I think it'd be nice just to access the fields on the config directly: 
```
    partition_dict = OrderedDict((key, typecast(key, config.selection[key][0])) for key in config.partition_keys)
    target = config.target_path.format(*partition_dict.values(), **partition_dict)
```